solutions/Makefile

```diff
@@ -42,8 +41,9 @@ $(DIRS):
 clean:
 	@ $(foreach i, $(DIRS), $(MAKE) -C $(i) clean $(NL) )
 
+# Run each solution tests, and clean each test after its completion to free up space 
 run:
-	@ $(foreach i, $(DIRS), $(MAKE) -C $(i) run $(NL) )
+	@ $(foreach i, $(DIRS), $(MAKE) -C $(i) run && $(MAKE) -C $(i) clean $(NL) )
```

Comment:
This would only make clean if the test passed. Though it makes a small impact, make clean should be done regardless test results. I would recommend using a bash for loop instead of the foreach function.

Reply:
I think it's reasonable to keep the file (don't run make clean) if a test failed so that we can investigate if we want.  And as for bash loop vs foreach, I think foreach is simpler to understand here, do I miss something?

Reply:
Actually, I'm going to go back on what I said last time as I did not consider others investigating manual builds. Cleanups are great on automated builds, but not on manual builds. This change would affect both.

@jxyang would you be fine with this clean up for solution tests during manual builds?

Reply:
I'd prefer the clean up is only activated for pipeline runs. See my comment above.

Is there an environment variable indicating the test/solution is being run on the pipeline? If not we should introduce one.

Reply:
That's acceptable to me. Maybe there is a way (defs.mak?) you can do that without touching hundreds of make files.